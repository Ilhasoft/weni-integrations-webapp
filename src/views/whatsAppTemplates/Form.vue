<template>
  <div class="templates-form">
    <div class="templates-form__content">
      <FormHeader />
      <FormTabs
        :formMode="this.formMode"
        :templateUuid="templateUuid"
        @manual-preview-update="updatePreview"
      />
    </div>

    <TemplatePreview :key="previewKey" class="templates-form__preview" />
  </div>
</template>

<script>
  import { mapActions } from 'pinia';
  import { whatsapp_store } from '@/stores/modules/appType/channels/whatsapp.store';
  import FormHeader from '@/components/whatsAppTemplates/FormHeader.vue';
  import FormTabs from '@/components/whatsAppTemplates/FormTabs.vue';
  import TemplatePreview from '@/components/whatsAppTemplates/TemplatePreview.vue';

  export default {
    name: 'WhatsAppTemplatesForm',
    components: {
      FormHeader,
      FormTabs,
      TemplatePreview,
    },
    data() {
      return {
        formMode: 'create',
        templateUuid: '',
        previewKey: 0,
      };
    },
    created() {
      const { templateUuid } = this.$route.params;

      if (templateUuid) {
        this.formMode = 'edit';
        this.templateUuid = templateUuid;
      }
    },
    unmounted() {
      this.clearAllTemplateFormData();
      this.clearTemplateData();
    },
    methods: {
      ...mapActions(whatsapp_store, ['clearAllTemplateFormData', 'clearTemplateData']),
      /* istanbul ignore next */
      updatePreview() {
        this.previewKey += 1;
      },
    },
  };
</script>

<style lang="scss" scoped>
  .templates-form {
    display: flex;
    height: 100%;

    &__content {
      flex: 1;
      max-width: calc(100vw - 310px);
      overflow: auto;
      display: flex;
      flex-direction: column;
    }

    &__preview {
      position: fixed;
      top: 0;
      right: 0;
      width: 270px;
      flex: 0 0 270px;
      height: 100vh;
      overflow: auto;
    }
  }
</style>
import { whatsapp_store } from '@/stores/modules/appType/channels/whatsapp.store'; whatsapp_store,
