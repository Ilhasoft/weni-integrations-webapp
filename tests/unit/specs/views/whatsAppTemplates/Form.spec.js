import Vuex from 'vuex';
import { mount, createLocalVue } from '@vue/test-utils';
import Form from '@/views/whatsAppTemplates/Form.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

const mountComponent = (routeParams = {}) => {
  const actions = {
    clearAllTemplateFormData: jest.fn(),
  };

  const store = new Vuex.Store({
    modules: {
      WhatsApp: {
        namespaced: true,
        actions,
      },
    },
  });

  const wrapper = mount(Form, {
    localVue,
    store,
    stubs: {
      FormHeader: true,
      FormTabs: true,
      TemplatePreview: true,
    },
    mocks: {
      $route: {
        params: routeParams,
      },
    },
  });

  return { wrapper, actions };
};

describe('views/whatsAppTemplates/Form.vue', () => {
  it('should be rendered properly', () => {
    const { wrapper } = mountComponent();
    expect(wrapper).toMatchSnapshot();
  });

  it('should set formMode to edit if templateUuid is provided in route', async () => {
    const { wrapper } = mountComponent({ templateUuid: '123' });
    expect(wrapper.vm.formMode).toEqual('edit');
  });

  it('should leave formMode as create edit if templateUuid is not provided in route', async () => {
    const { wrapper } = mountComponent();
    expect(wrapper.vm.formMode).toEqual('create');
  });

  describe('destroyed()', () => {
    it('should call clearAllTemplateFormData', async () => {
      const { wrapper, actions } = mountComponent();

      expect(actions.clearAllTemplateFormData).not.toHaveBeenCalled();
      wrapper.destroy();
      expect(actions.clearAllTemplateFormData).toHaveBeenCalledTimes(1);
    });
  });
});
