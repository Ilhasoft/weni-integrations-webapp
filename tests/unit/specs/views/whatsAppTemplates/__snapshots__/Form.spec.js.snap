// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/whatsAppTemplates/Form.vue should be rendered properly 1`] = `
<div class="templates-form">
  <div class="templates-form__content">
    <formheader-stub></formheader-stub>
    <formtabs-stub formmode="create" templateuuid=""></formtabs-stub>
  </div>
  <templatepreview-stub showtitle="true" class="templates-form__preview"></templatepreview-stub>
</div>
`;
